cmd/types.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/artifacts"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/pyxis"
+	log "github.com/sirupsen/logrus"
+)
+
+// resultWriter defines methods associated with writing check results.
+type resultWriter interface {
+	OpenFile(name string) (io.WriteCloser, error)
+	io.WriteCloser
+}
+
+// resultSubmitter defines methods associated with submitting results to Red HAt.
+type resultSubmitter interface {
+	Submit(context.Context) error
+}
+
+// pyxisClient defines pyxis API interactions that are relevant to check executions in cmd.
+type pyxisClient interface {
+	FindImagesByDigest(ctx context.Context, digests []string) ([]pyxis.CertImage, error)
+	GetProject(context.Context) (*pyxis.CertProject, error)
+	SubmitResults(context.Context, *pyxis.CertificationInput) (*pyxis.CertificationResults, error)
+}
+
+// newPyxisClient initializes a pyxisClient with relevant information from cfg.
+// If the the CertificationProjectID, PyxisAPIToken, or PyxisHost are empty, then nil is returned.
+// Callers should treat a nil pyxis client as an indicator that pyxis calls should not be made.
+func newPyxisClient(ctx context.Context, cfg certification.Config) pyxisClient {
+	if cfg.CertificationProjectID() == "" || cfg.PyxisAPIToken() == "" || cfg.PyxisHost() == "" {
+		return nil
+	}
+
+	return pyxis.NewPyxisClient(
+		cfg.PyxisHost(),
+		cfg.PyxisAPIToken(),
+		cfg.CertificationProjectID(),
+		&http.Client{Timeout: 60 * time.Second},
+	)
+}
+
+// containerCertificationSubmitter submits container results to Pyxis, and implements
+// a resultSubmitter.
+type containerCertificationSubmitter struct {
+	certificationProjectID string
+	pyxis                  pyxisClient
+	dockerConfig           string
+	preflightLogFile       string
+}
+
+func (s *containerCertificationSubmitter) Submit(ctx context.Context) error {
```

Comment:
There seems to be at least [this](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/cmd/check_container.go#L252) missing in this function. I'm not sure if other things are missed as well.

Reply:
That's not part of the submit function, which is why it's not included here.

Reply:
Yeah I noticed that in the other PR. Hard to follow all of these, just want to make sure we don't miss something.